_studio/hevce_hw/CMakeLists.txt

```diff
@@ -34,7 +34,7 @@ else()
   set( hevc_version $ENV{MFX_HEVC_VERSION} )
 endif()
 
-if( __ARCH MATCHES intel64 )
+if(CMAKE_SYSTEM_PROCESSOR MATCHES x86_64)
```

Comment:
better use:
````CMAKE
if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
````

To make sure **CMake** hits a 64bits architecture.

Reply:
How this is better and why? CMAKE_SYSTEM_PROCESSOR is a variable which can be overwritten in cross-compiling to make sure we build for the right architecture. CMAKE_SIZEOF_VOID_P corresponds to target architecture or to host architecture or to what?

Reply:
How to make sure that any of the "64 bits" archs  that may exist will match the expression "x86_64"  ? This is not so obvious.
CMAKE_SIZEOF_VOID_P is computed from a test-run; It's what the compiler told to cmake.